js/common/lib/training-session.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {InferenceSession} from './inference-session.js';
+import {TrainingSession as TrainingSessionImpl} from './training-session-impl.js';
+
+/* eslint-disable @typescript-eslint/no-redeclare */
+
+/**
+ * Represent a runtime instance of an ONNX model.
+ */
+export interface TrainingSession {
+  // #region release()
+
+  /**
+   * Release the inference session and the underlying resources.
+   */
+  release(): Promise<void>;
+  // #endregion
+
+  // #region metadata
+
+  /**
+   * Get input names of the loaded model.
+   */
+  readonly inputNames: readonly string[];
+
+  /**
+   * Get output names of the loaded model.
+   */
+  readonly outputNames: readonly string[];
+  // #endregion
+}
```

Comment:
what can a training session do? seems missing something in the interface?

Reply:
I previously included runTrainStep in the interface & had the method implementation throw an error, but the implementation produced a lot of linting issues so I removed the method from this PR -- I'm open to adding it in again though, and I will add more documentation on the TrainingSession.